File: frontend/types/link.ts
// ===== link 模块类型（与 backend schemas/link.py 对齐）=====

/** 链接文件大小单位枚举（与后端 SizeUnitEnum 对齐） */
export type SizeUnit = 'KB' | 'MB' | 'GB'

export interface LinkItem {
  id: number
  url: string
  content: string | null
  code: string | null
  category: string | null
  /** 链接大小数值（浮点），配合 size_unit 使用；旧数据 size_unit 为 null 时按字节处理 */
  size: number | null
  /** 大小单位：KB / MB / GB，旧数据可能为 null */
  size_unit: SizeUnit | null
  status: boolean
  /** 所属游戏 id：按用户过滤的链接列表据此跳转对应游戏/管理页 */
  game_id: number
}

export interface LinkItems {
  item: LinkItem
  account: {
    username: string
    avatar: string | null
  }
}

/** GET /api/v1/link/{game_id} 列表响应 */
export interface LinkList {
  total: number
  items: LinkItems[]
}

/** POST /api/v1/link/add 请求体 */
export interface AddLink {
  url: string
  content?: string | null
  code?: string | null
  category?: string | null
  size?: number | null
  size_unit?: SizeUnit | null
}

/** PUT /api/v1/link/review 请求体（全部可选） */
export interface EditLink {
  content?: string | null
  code?: string | null
  category?: string | null
  size?: number | null
  size_unit?: SizeUnit | null
}

File: backend/app/schemas/link.py
from pydantic import BaseModel, Field
from typing import List, Optional
from .user import UserInfo


class LinkItem(BaseModel):
    """链接响应模型：单条链接信息（含审核状态与所属游戏 id）"""
    id: int
    url: str
    content: Optional[str] = None
    code: Optional[str] = None
    category: Optional[str] = None
    size: Optional[float] = None
    size_unit: Optional[str] = None
    status: bool
    # 所属游戏 id：按用户过滤的链接列表需要据此跳转到对应游戏/管理页
    game_id: int


class AddLink(BaseModel):
    """新增链接请求模型：服务端 add 时强制 status=True 直接公开，故请求体不含 status"""
    url: str = Field(min_length=1, max_length=255)
    content: Optional[str] = Field(default=None, max_length=255)
    code: Optional[str] = Field(default=None, max_length=255)
    category: Optional[str] = Field(default=None, max_length=255)
    size: Optional[float] = Field(default=None, ge=0)
    size_unit: Optional[str] = Field(default=None, pattern="^(KB|MB|GB)$")


class EditLink(BaseModel):
    """编辑链接请求模型：仅允许更新链接内容字段，全部可选"""
    content: Optional[str] = Field(default=None, max_length=255)
    code: Optional[str] = Field(default=None, max_length=255)
    category: Optional[str] = Field(default=None, max_length=255)
    size: Optional[float] = Field(default=None, ge=0)
    size_unit: Optional[str] = Field(default=None, pattern="^(KB|MB|GB)$")


class LinkItems(BaseModel):
    item: LinkItem
    account: UserInfo


class LinkList(BaseModel):
    total: int
    items: List[LinkItems]
